--- config.mk	2023-11-12 12:13:38.827565454 +0000
+++ config2.mk	2023-11-12 12:14:57.474568619 +0000
@@ -12,8 +12,8 @@ X11LIB = /usr/X11R6/lib
 
 # flags
 CPPFLAGS = -I$(X11INC) -D_DEFAULT_SOURCE -DVERSION=\"${VERSION}\"
-CFLAGS   = -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter -Os
-LDFLAGS  = -L$(X11LIB) -s
+CFLAGS   = -march=native -flto -O3 -fno-plt -ffunction-sections -fdata-sections ${INCS} ${CPPFLAGS}
+LDFLAGS  = -L$(X11LIB) -s -Wl,--gc-sections -flto
 # OpenBSD: add -lsndio
 # FreeBSD: add -lkvm -lsndio
 LDLIBS   = -lX11
