--- dwm.c	2021-07-11 22:17:10.902666299 +0100
+++ dwm_new.c	2021-07-11 22:18:21.886002158 +0100
@@ -200,6 +200,7 @@
 static void focus(Client *c);
 static void focusin(XEvent *e);
 static void focusmon(const Arg *arg);
+static void focusurgent(const Arg *arg);
 static void focusstack(const Arg *arg);
 static Atom getatomprop(Client *c, Atom prop);
 static int getrootptr(int *x, int *y);
@@ -1104,6 +1105,21 @@
 }
 
 void
+focusurgent(const Arg *arg) {
+	Client *c;
+	int i;
+	for(c=selmon->clients; c && !c->isurgent; c=c->next);
+	if(c) {
+		for(i=0; i < LENGTH(tags) && !((1 << i) & c->tags); i++);
+		if(i < LENGTH(tags)) {
+			const Arg a = {.ui = 1 << i};
+			view(&a);
+			focus(c);
+		}
+	}
+}
+
+void
 focusstack(const Arg *arg)
 {
 	Client *c = NULL, *i;
