--- dwm.c	2021-10-23 18:56:23.256679223 +0100
+++ dwm_new.c	2021-10-23 18:57:16.255075262 +0100
@@ -254,6 +254,7 @@ static void tagmon(const Arg *arg);
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
+static void togglefullscreen(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
 static void unfocus(Client *c, int setfocus);
@@ -2310,6 +2311,13 @@ togglefloating(const Arg *arg)
 	arrange(selmon);
 }
 
+togglefullscreen(const Arg *arg) {
+    if (!selmon->sel)
+        return;
+
+    setfullscreen(selmon->sel, !selmon->sel->isfullscreen);
+}
+
 void
 toggletag(const Arg *arg)
 {
